### PR TITLE
shrapnel for improvised explosives: 3rd times the charm

### PR DESCRIPTION
**Pull request by ultraplen**
The 3rd PR for this, this time it seems to work. I literally could not unfuck my last 2 PRs. 1st pr : #12961

@9600bauds @PJB3005 and @Exxion had open reviews on the first PR.

I accidentaly some sharpness flags but I manually readded them, should be fine.

Copy paste:
In reponse to #12398

Allows players to add any w_class_tiny item to an improvised explosive. The items will then launch out in random directions when it explodes. Some items, not necessarily tiny, will disintegrate into shrapnel which deals spicy damage. The code makes it semi simple to add custom types of shrapnel for certain items in the future.

The numbers for those who do not want to read through the code:

 * 8 item slots in each explosive. (Previously 10) Shards take up 2 slots (Previously 1)

 * Shards disintegrate into 3 pieces of small shrapnel (Previously 5 normal sized)

 * Standard shrapnel deals 45 damage and small shrapnel 25. Shrapnel is child of projectile/bullet (Previously 55 damage for standard shrapnel)

 * Shrapnel targets a random bodypart

 * Plasma shards produce purple TOX damage (35) shrapnel

 * Kitchen utensils produce 1 standard shrapnel and take up 2 slots each

 * Individual pieces of shrapnel have a random range limit of 6 to 10 squares

 * Bullet casings fire their projectile if its not already spent.

 * All spent bullet casings are small shrapnel.

As it is now this is fairly balanced tbh. Its a buff to the IED for sure but it was trash earlier.

Clip of some crayons and 5 shards: https://puu.sh/sVJ5o.mp4

:cl:

 * rscadd: Shrapnel, in the form of tiny items and shards, can be added to improvised explosives.

### CODE REVIEW

**Comment by Probe1** [comment]
look at this until the feeling of frustration and/or 😡 passes http://www.cutestpaw.com/wp-content/uploads/2014/08/corgi.jpg

File: code/game/objects/items/weapons/grenades/ghettobomb.dm [hunk 1]
```diff
@@ -61,6 +63,27 @@
...
+	if(assembled == 2 && shrapnel_list.len > 0)
+
+
+		to_chat(usr, "<span  class='notice'>You remove all the shrapnel from the improvised explosive.</span>")
+		current_shrapnel = 0
+		for(var/obj/item/shrapnel in shrapnel_list)
+
+			shrapnel.forceMove(get_turf(src))
+			shrapnel_list.Remove(shrapnel)
+		current_shrapnel = 0
```
Comment (ComicIronic) [inline code/game/objects/items/weapons/grenades/ghettobomb.dm:85]: You set this twice.

File: code/game/objects/items/weapons/grenades/ghettobomb.dm [hunk 2]
```diff
@@ -82,9 +105,28 @@
 			spawn(det_time)
 				prime()
 
+
+/obj/item/weapon/grenade/iedcasing/proc/AddShrapnel(var/obj/item/I, mob/user as mob)
```
Comment (ComicIronic) [inline code/game/objects/items/weapons/grenades/ghettobomb.dm]: You should really use snake_case to be consistent with the rest of the file and most of the codebase.

Comment (ultraplen) [reply code/game/objects/items/weapons/grenades/ghettobomb.dm]: snake what ?

Comment (ultraplen) [reply code/game/objects/items/weapons/grenades/ghettobomb.dm]: oh, you mean the name ?

Comment (ComicIronic) [reply code/game/objects/items/weapons/grenades/ghettobomb.dm]: Snake case. It's a naming standard for vars and procs that produces `stuff_like_this`. What you've used is Pascal case, which does `StuffLikeThis`.

**Review by ComicIronic — APPROVED** [review]

**Comment by Probe1** [comment]
whats the worst that can happen (runtimes server crashes everyone hates us)

**Comment by SonixApache** [comment]
Wow, this lets you make bullet bombs? That's fucking awesome.

**Comment by ultraplen** [comment]
Hell yeah dude

**Pull request merged**